test/e2e/data/e2e_conf.yaml

```diff
@@ -228,9 +228,9 @@ providers:
           - sourcePath: "./infrastructure-aws/generated/cluster-template-upgrades.yaml"
           - sourcePath: "./infrastructure-aws/generated/cluster-template-peered-remote.yaml"
           - sourcePath: "./infrastructure-aws/generated/cluster-template-internal-elb.yaml"
-          - sourcePath: "./infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml"
           - sourcePath: "./infrastructure-aws/generated/cluster-template-nested-multitenancy-clusterclass.yaml"
```

Comment:
Since we are having a separate ClusterClass manifests folder, how about keeping everything related to those manifests in the same folder, i.e., moving its kustomize files and generating those with a new target `make generate-clusteclass-flavors`?

And eventually, we will have a base ClusterClass and possibly base InfraTemplates similar to base Cluster kustomize file to reuse those across different flavors.

Reply:
Sure. What should be the new directory structure? Should it be sth like this?

Current
```
|- infrastructure-aws
    |--generated
    |--kustomized_sources
```

New
```
|- infrastructure-aws
    |--generated
    |--kustomized_sources
    |--topology
        |--generated
        |--kustomized_sources
        |--clusterclasses <- clusterclasses still need to be a separate dir as they are not generated by kustomize
```

Reply:
Yes, modified little bit:
```
|- infrastructure-aws
    |--withoutclusterclass (with better naming)
        |--generated
        |--kustomized_sources
    |--withclusterclass
        |--generated
        |--kustomized_sources
           |--clusterclassbase
           |--awsclustertemplatebase
           |--multitenancy-flavor
```

Reply:
Refactored the directory structure. 

I didn't create awsclustertemplatebase dir yet as I am not sure if we want to create base directories for all the templates (like kcpbase, awsmachinebase and etc) but that will be an easy change to do if we want to do it later. Trying to understand what the best practices in the community.